### ISSUE TITLE
Wrong cart total value

### ISSUE THREAD

**Issue by fmcdigital** [darryldecode/laravelshoppingcart#74]
I'm having a problem when get the cart Totals.

For example: I have an item that costs 2399, when I get Totals it returns 2.

And prices like 19.90 is returning 20...

**Comment by it-can** [comment]
same problem here... I have a product costing 1,06, when I add 4, it will output 4 instead of 4,24. It now does some rounding up somewhere...

**Comment by ebisbe** [comment]
Add SHOPPING_DECIMALS=2 to set the number of decimals you want to use. By default is 0 now. I have to write the documentation.

**Comment by fmcdigital** [comment]
Thanks @ebisbe!

**Comment by it-can** [comment]
@ebisbe does not work for me, tried that...

**Comment by ebisbe** [comment]
@it-can ok, let me check it to find the problem.

**Comment by it-can** [comment]
@ebisbe it seems this line could be the problem
https://github.com/darryldecode/laravelshoppingcart/blob/master/src/Darryldecode/Cart/Cart.php#L585

It gets the formatted value, in my case with a comma, it will me screwed up... Why not add a format function, so that the underlying functions will use floats instead of formatted values?

**Comment by ebisbe** [comment]
Yes the issue is there. But adding a format function what will solve? The problem there is that getTotal() uses getSubtotal(). For a faster solution I would add a bool parameter in the getSubtotal to where return the value formated or not. 
But I think the best option is trying to use some sort of presenters for the values. That way instead of calling ->getTotal() we would call ->total ( using the formaters )

**Comment by it-can** [comment]
Isnt it possible to do something like:

```
$cart->getTotal()->format();
```

**Comment by ebisbe** [comment]
Yes it's possible but it's ugly. The getTotal() it's suppossed to be used to print the value. So you always want it formated. I don't see when we want to use just $cart->getTotal()

**Comment by it-can** [comment]
Personally I would rather format it myself, so it would be nice to have a option to get the float...

**Comment by ebisbe** [comment]
Why would you format it yourself when you can do it inside the package and avoid handle it outside? Just explain me to understand your concerns. 
Another option is to add another config option to use format a bool value.

**Comment by fmcdigital** [comment]
I have a custom global function to format all my values(prices),

```
function str_price($val)
{
    $price = number_format($val, 2, ',', '.');
    return 'R$ ' . $price;
}
```

and then you call,

```
str_price(Cart::getTotal())
```

**Comment by it-can** [comment]
@fmcdigital that's what I also do, I want to have full control over the formatting of money values. I want to be flexible.

**Comment by ebisbe** [comment]
But @fmcdigital and @it-can you are flexible with my solution.  This should do the trick https://github.com/darryldecode/laravelshoppingcart/pull/75 but you have to let the package handle all the values. 

I think is better to use only 
``Cart:getTotal()``
than 
``str_price(Cart::getTotal())``

**Comment by ebisbe** [comment]
@fmcdigital what 'R' stands for? I understand the dollar sign but not the R

**Comment by it-can** [comment]
@ebisbe what about this file?

https://github.com/darryldecode/laravelshoppingcart/blob/f851e2ab538880a90b408306488effbb8056e1de/src/Darryldecode/Cart/ItemCollection.php#L128

**Comment by ebisbe** [comment]
same issue

**Comment by fmcdigital** [comment]
@ebisbe I know, but I use ``Cart::getTotal()`` when I send to my payment API and some sums, but the str_price function is only to display in my frontend.

The 'R' is Brazilian Real -> R$ :)

**Comment by ebisbe** [comment]
ok, I add an option to disable it. So it maintains backwards compatibility. This would be fine then

**Comment by fmcdigital** [comment]
And for me is really ok to use my format function to display on frontend, I can use that for shipping methods to

**Comment by it-can** [comment]
@darryldecode what is your view on this?

**Comment by ebisbe** [comment]
@it-can @fmcdigital  Please review the #75

**Comment by fmcdigital** [comment]
@ebisbe the #75 is a good solution, but as I said, my controllers gets the raw value and the formats are only used on frontend for my customers.

**Comment by ebisbe** [comment]
@fmcdigital With this PR is disabled by default. So it should be compatible with your code.

**Comment by it-can** [comment]
Nice work

**Comment by lvian** [comment]
Just passing by to thank you guys for adding the backward compatibility, I too suddenly had a website stop working because of the change :)

**Comment by ebisbe** [comment]
@lvian Sorry for that. I'm still thinking that the solution is not the best. And the one proposed @it-can is better solution... Just for now works for everyone ( I hope )
